### PR TITLE
Fix wire compatibility tests, decode result from go protobuf, use the new protobuf

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: consumer/pdata/metric_test.go [hunk 1]
```diff
@@ -18,9 +18,9 @@ import (
 	"testing"
 
 	gogoproto "github.com/gogo/protobuf/proto"
-	goproto "github.com/golang/protobuf/proto" //lint:ignore SA1019 golang/protobuf/proto is deprecated
-	otlpmetricsgoproto "github.com/open-telemetry/opentelemetry-proto/gen/go/metrics/v1"
```
Comment (tigrannajaryan) [inline consumer/pdata/metric_test.go:22]: Is this change needed to stop importing `github.com/open-telemetry/opentelemetry-proto/gen` (which no longer exists)

File: consumer/pdata/metric_test.go [hunk 2]
```diff
@@ -39,18 +39,18 @@ func TestResourceMetricsWireCompatibility(t *testing.T) {
 	assert.NotNil(t, wire1)
 
 	// Unmarshal from the wire to OTLP Protobuf in goproto's representation.
-	var goprotoRM otlpmetricsgoproto.ResourceMetrics
-	err = goproto.Unmarshal(wire1, &goprotoRM)
+	var goprotoMessage emptypb.Empty
```
Comment (tigrannajaryan) [inline consumer/pdata/metric_test.go:42]: I didn't know about `emptypb.Empty`. So this allows representing any message and fields are just stored as unknown fields?

Comment (bogdandrutu) [reply consumer/pdata/metric_test.go:42]: Yes. We can also if you want generate the protos with Google protobuf in the future. But the otel-proto are removed anyway so for the next update this is important to change

File: consumer/pdata/metric_test.go (hunk 1 above)
Comment (bogdandrutu) [reply consumer/pdata/metric_test.go:22]: Yes, remove the warning

File: consumer/pdata/metric_test.go (hunk 2 above)
Comment (tigrannajaryan) [reply consumer/pdata/metric_test.go:42]: No, this is good.
